host/src/c_api/c_api.cc

```diff
@@ -848,6 +848,10 @@ XGB_DLL int XGBCreateEnclave(const char *enclave_image, uint32_t flags, int log_
   if (!Enclave::getInstance().getEnclave()) {
     oe_result_t result;
 
+#ifdef __ENCLAVE_SIMULATION__
+    flags |= OE_ENCLAVE_FLAG_SIMULATE;
```

Comment:
Now that we automatically add the `SIMULATE` flag, in Python should we expose flags to the user? Right now the user can pass in any combination of `RELEASE`, `DEBUG`, or `SIMULATE`.  

In my opinion we should probably take out the `SIMULATE` flag from all examples, since that flag is now added automatically. What are your thoughts on keeping the `DEBUG` / `RELEASE` mode flags exposed to the user? 

It seems we have two options:

1. Since the user will have to specify whether to run in debug or release mode when building (by presumably modifying the root CMakeLists.txt), we can only create enclaves in whatever mode the user specifies, i.e. if the user specifies `debug = 0` when building we can only create enclaves in release mode. I think this would require another compile time flag, similar to `__ENCLAVE_SIMULATION__`. However, this approach is a bit inflexible, since the user would not be able to run in release mode when specifying `debug = 1` when building.

2. The other option is to keep as is, i.e. expose the `DEBUG` and `RELEASE` flags to the user. This approach is a bit more flexible but gives the user one more thing to think about.

Reply:
I like the idea of not exposing these flags to the user at all. 
Is there a situation possible where the runtime flags can differ from the build? If not, then exposing these flags to the user is pointless. However, if there is a way for the user to override the build options at runtime, then we might want to consider leaving it be.

Reply:
The only situation I can think of is when the user builds everything in debug mode but then runs with the `RELEASE` mode flag. I don't really see any reason to do this though, since it'll still be insecure because the enclave was built with debugging enabled. So maybe we should just not expose the flags at all, and instead use MACROS that are defined depending on how the user builds chooses to build the enclave?

Reply:
Ok. I think it's safe to remove the user-facing flags from the demos. Will update.

Reply:
Ok, I think it's safe to remove the flag from the demos. Will update.

Reply:
Fixed!

Reply:
Fixed!